src/LibraryManager.Vsix/UI/Controls/Library.xaml.cs

```diff
@@ -284,5 +291,39 @@ private void Library_LostFocus(object sender, RoutedEventArgs e)
                 Flyout.IsOpen = false;
             }
         }
+
+        private string GetLibraryName(string insertionText)
```

Comment:
If I understand correctly, this value gets appended to the `_baseFolder` in `TargetLocation.xaml.cs` 
If that is the case, you should be calling the provider to get the suggested folder name. 

The current logic doesn't work well for `unpkg` and `filesystem` providers.

For eg: `unpkg` allows library Ids like `@angular/cli@4.0.0` , the current logic would return empty string. 
For `filesystem` it would end up returning the entire path of the library (if it doesn't contain `@`)

Reply:
I am start searching the index of @ from the second character so I think it would be Okay for unpkg (index will be -1 if it's not completed). If it was filesystem provider (at index == -1), I will return the insertiontext.

---
In reply to: [200735579](https://github.com/aspnet/LibraryManager/pull/285#discussion_r200735579) [](ancestors = 200735579)

Reply:
Now changed to get the value from InstallDialogViewModel.cs, so delete the function.

---
In reply to: [200749858](https://github.com/aspnet/LibraryManager/pull/285#discussion_r200749858) [](ancestors = 200749858,200735579)